### PR TITLE
text spotting accuracy checking

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
@eaidova

File: tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn_with_text.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                description='Name of output layer with texts'
+            ),
+            'confidence_threshold': NumberField(
+                description='Confidence threshold that is used to filter out detected instances'
+            ),
+            'num_detections_out': StringField(
+                optional=True, description='Name of output layer with number valid detections '
+                                           '(used in MaskRCNN models trained with TF Object Detection API)'
+            ),
+            'detection_out': StringField(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn_with_text.py]: do you really need `detection_out` and `num_detections_out`?
MaskRCNN adapter is very common entity, where depends from source framework we have different outputs with respective processing.
I think you can stay only necessary for your model outputs.

File: tools/accuracy_checker/accuracy_checker/representation/text_detection_representation.py [hunk 2]
```diff
@@ -58,5 +58,5 @@ def remove(self, indexes):
         self.description = np.delete(self.description, indexes)
 
 
-class TextDetectionPrediction(TextDetectionRepresentation):
+class TextDetectionPrediction(TextDetectionAnnotation):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/representation/text_detection_representation.py]: please stay base class `TextDetectionRepresentation`, just move description to the base class.
inheritance is now logically incorrect, predicted object should no be treated as annotation

File: tools/accuracy_checker/configs/text-spotting-0001.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            raw_masks_out: raw_masks
+            texts_out: texts
+            inputs:
+                -   name: im_info
+                    type: IMAGE_INFO
+            confidence_threshold: 0.8
+
+      launchers:
+        - framework: dlsdk
+          device: CPU
```
Comment (eaidova) [inline tools/accuracy_checker/configs/text-spotting-0001.yml]: please remove device, we prefer to change it in command line

File: tools/accuracy_checker/configs/text-spotting-0001.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        eos_index: '1'
+
+        adapter:
+            type: mask_rcnn_with_text
+            classes_out: classes
+            scores_out: scores
+            boxes_out: boxes
+            raw_masks_out: raw_masks
+            texts_out: texts
+            inputs:
```
Comment (eaidova) [inline tools/accuracy_checker/configs/text-spotting-0001.yml]: why are inputs located inside adapter?

File: tools/accuracy_checker/custom_evaluators/text_spotting_evaluator.py [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+            launcher.create_ie_plugin()
+        self.exec_network = launcher.plugin.load(self.network)
+        self.input_blob = next(iter(self.network.inputs))
+
+    def predict(self, identifiers, input_data):
+        input_data = np.array(input_data)
+        assert len(input_data.shape) == 4
+        assert input_data.shape[0] == 1
+
+        input_data = {'im_data': self.fit_to_input(input_data),
```
Comment (eaidova) [inline tools/accuracy_checker/custom_evaluators/text_spotting_evaluator.py]: it can be dangerous to hardcode names of inputs here...

**Comment by eaidova** [comment]
jenkins ok to test

File: tools/accuracy_checker/custom_evaluators/text_spotting_evaluator.py [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+    if not model_class:
+        raise ValueError('model for framework {} is not supported'.format(framework))
+    return model_class(model_config, launcher)
+
+
+class SequentialModel(BaseModel):
+    def __init__(self, network_info, launcher):
+        super().__init__(network_info, launcher)
+        if not contains_all(network_info, ['detector', 'recognizer_encoder', 'recognizer_decoder']):
+            raise ConfigError('network_info should contains encoder and decoder fields')
```
Comment (eaidova) [inline tools/accuracy_checker/custom_evaluators/text_spotting_evaluator.py]:
```suggestion
            raise ConfigError('network_info should contains detector, encoder and decoder fields')
```

**Comment by Ilya-Krylov** [comment]
@eaidova

File: tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn_with_text.py [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+            ),
+            'scores_out': StringField(
+                description="Name of output layer with bbox scores."
+                            "(optional, if your model has detection_output layer as output).",
+                optional=True
+            ),
+            'boxes_out': StringField(
+                description="Name of output layer with bboxes."
+                            "(optional, if your model has detection_output layer as output).",
+                optional=True
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn_with_text.py]: you can made outpus required (`optional=False`), it eliminates the need to check they in `configure()`

File: tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn_with_text.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+        self.scores_out = self.get_value_from_config('scores_out')
+        self.boxes_out = self.get_value_from_config('boxes_out')
+        self.num_detections_out = self.get_value_from_config('num_detections_out')
+
+        self.raw_masks_out = self.get_value_from_config('raw_masks_out')
+        self.texts_out = self.get_value_from_config('texts_out')
+
+        self.confidence_threshold = self.get_value_from_config('confidence_threshold')
+
+        self.realisation = self._process_pytorch_outputs
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn_with_text.py]: you do not have any other realisation, so you can call this method directly

File: tools/accuracy_checker/custom_evaluators/text_spotting_evaluator.py [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+class DetectorDLSDKModel(BaseModel):
+    def __init__(self, network_info, launcher):
+        super().__init__(network_info, launcher)
+        model_xml = str(network_info['model'])
+        model_bin = str(network_info['weights'])
+        self.network = launcher.create_ie_network(model_xml, model_bin)
+        if not hasattr(launcher, 'plugin'):
+            launcher.create_ie_plugin()
+        self.exec_network = launcher.plugin.load(self.network)
+        self.im_info_name = \
```
Comment (eaidova) [inline tools/accuracy_checker/custom_evaluators/text_spotting_evaluator.py]: we avoid using `\` for moving to the next line, please use brackets

**Comment by eaidova** [comment]
@Ilya-Krylov, please update README.md for annotation converters and adapters for new features

**Pull request merged**